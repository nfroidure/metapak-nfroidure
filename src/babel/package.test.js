'use strict';

const packageTransformer = require('./package');

describe('Babel', () => {
  describe('Package transformer', () => {
    it('should work with some files', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
            preversion: '',
          },
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
            configs: ['babel'],
          },
        })
      ).toMatchSnapshot();
    });

    it('should work with mocha config', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
            preversion: '',
          },
          devDependencies: {},
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
            configs: ['babel', 'mocha'],
          },
        })
      ).toMatchSnapshot();
    });

    it('should work with jest config', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
            preversion: '',
          },
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
            configs: ['babel', 'jest'],
          },
        })
      ).toMatchSnapshot();
    });

    it('should work with eslint config', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
            preversion: '',
          },
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
            configs: ['eslint', 'babel'],
          },
        })
      ).toMatchSnapshot();
    });
  });
});
