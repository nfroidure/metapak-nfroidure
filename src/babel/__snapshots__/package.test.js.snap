// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel Package transformer should work with child packages 1`] = `
Object {
  "babel": Object {
    "env": Object {
      "cjs": Object {
        "comments": true,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": "commonjs",
              "targets": Object {
                "node": "10",
              },
            },
          ],
        ],
      },
      "mjs": Object {
        "comments": false,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": false,
              "targets": Object {
                "node": "12",
              },
            },
          ],
        ],
      },
    },
    "plugins": Array [
      "@babel/plugin-proposal-object-rest-spread",
    ],
    "presets": Array [
      Array [
        "@babel/env",
        Object {
          "targets": Object {
            "node": "12.19.0",
          },
        },
      ],
    ],
    "sourceMaps": true,
  },
  "devDependencies": Object {
    "@babel/cli": "^7.10.5",
    "@babel/core": "^7.11.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.11.0",
    "@babel/preset-env": "^7.11.0",
    "@babel/register": "^7.10.5",
  },
  "main": "index",
  "metapak": Object {
    "data": Object {
      "bundleFiles": Array [
        "dist",
        "src",
      ],
      "childPackage": true,
      "files": "src/*.js src/**/*.js",
    },
  },
  "module": "index.mjs",
  "scripts": Object {
    "compile": "rimraf -f 'dist' && npm run compile:cjs && npm run compile:mjs",
    "compile:cjs": "babel --env-name=cjs --out-dir=dist --source-maps=true src",
    "compile:mjs": "babel --env-name=mjs --out-file-extension=.mjs --out-dir=dist --source-maps=true src",
  },
}
`;

exports[`Babel Package transformer should work with eslint config 1`] = `
Object {
  "babel": Object {
    "env": Object {
      "cjs": Object {
        "comments": true,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": "commonjs",
              "targets": Object {
                "node": "10",
              },
            },
          ],
        ],
      },
      "mjs": Object {
        "comments": false,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": false,
              "targets": Object {
                "node": "12",
              },
            },
          ],
        ],
      },
    },
    "plugins": Array [
      "@babel/plugin-proposal-object-rest-spread",
    ],
    "presets": Array [
      Array [
        "@babel/env",
        Object {
          "targets": Object {
            "node": "12.19.0",
          },
        },
      ],
    ],
    "sourceMaps": true,
  },
  "devDependencies": Object {
    "@babel/cli": "^7.10.5",
    "@babel/core": "^7.11.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.11.0",
    "@babel/preset-env": "^7.11.0",
    "@babel/register": "^7.10.5",
    "babel-eslint": "^10.1.0",
  },
  "eslintConfig": Object {
    "env": Object {
      "es6": true,
      "jest": true,
      "mocha": true,
      "node": true,
    },
    "extends": "eslint:recommended",
    "parserOptions": Object {
      "ecmaVersion": 2018,
      "modules": true,
      "sourceType": "module",
    },
    "plugins": Array [
      "prettier",
    ],
    "rules": Object {
      "prettier/prettier": "error",
    },
  },
  "greenkeeper": Object {
    "ignore": Array [
      "@babel/cli",
      "@babel/core",
      "@babel/register",
      "@babel/preset-env",
      "@babel/plugin-proposal-object-rest-spread",
      "babel-eslint",
    ],
  },
  "main": "index",
  "metapak": Object {
    "configs": Array [
      "eslint",
      "babel",
    ],
    "data": Object {
      "bundleFiles": Array [
        "dist",
        "src",
      ],
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "module": "index.mjs",
  "prettier": Object {
    "printWidth": 80,
    "proseWrap": "always",
    "semi": true,
    "singleQuote": true,
    "trailingComma": "all",
  },
  "scripts": Object {
    "compile": "rimraf -f 'dist' && npm run compile:cjs && npm run compile:mjs",
    "compile:cjs": "babel --env-name=cjs --out-dir=dist --source-maps=true src",
    "compile:mjs": "babel --env-name=mjs --out-file-extension=.mjs --out-dir=dist --source-maps=true src",
    "precz": "npm run compile",
    "preversion": "npm run compile",
    "test": "",
  },
}
`;

exports[`Babel Package transformer should work with jest config 1`] = `
Object {
  "babel": Object {
    "env": Object {
      "cjs": Object {
        "comments": true,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": "commonjs",
              "targets": Object {
                "node": "10",
              },
            },
          ],
        ],
      },
      "mjs": Object {
        "comments": false,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": false,
              "targets": Object {
                "node": "12",
              },
            },
          ],
        ],
      },
    },
    "plugins": Array [
      "@babel/plugin-proposal-object-rest-spread",
    ],
    "presets": Array [
      Array [
        "@babel/env",
        Object {
          "targets": Object {
            "node": "12.19.0",
          },
        },
      ],
    ],
    "sourceMaps": true,
  },
  "devDependencies": Object {
    "@babel/cli": "^7.10.5",
    "@babel/core": "^7.11.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.11.0",
    "@babel/preset-env": "^7.11.0",
    "@babel/register": "^7.10.5",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "@babel/cli",
      "@babel/core",
      "@babel/register",
      "@babel/preset-env",
      "@babel/plugin-proposal-object-rest-spread",
      "babel-core",
    ],
  },
  "main": "index",
  "metapak": Object {
    "configs": Array [
      "babel",
      "jest",
    ],
    "data": Object {
      "bundleFiles": Array [
        "dist",
        "src",
      ],
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "module": "index.mjs",
  "scripts": Object {
    "compile": "rimraf -f 'dist' && npm run compile:cjs && npm run compile:mjs",
    "compile:cjs": "babel --env-name=cjs --out-dir=dist --source-maps=true src",
    "compile:mjs": "babel --env-name=mjs --out-file-extension=.mjs --out-dir=dist --source-maps=true src",
    "precz": "npm run compile",
    "preversion": "npm run compile",
    "test": "",
  },
}
`;

exports[`Babel Package transformer should work with mocha config 1`] = `
Object {
  "babel": Object {
    "env": Object {
      "cjs": Object {
        "comments": true,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": "commonjs",
              "targets": Object {
                "node": "10",
              },
            },
          ],
        ],
      },
      "mjs": Object {
        "comments": false,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": false,
              "targets": Object {
                "node": "12",
              },
            },
          ],
        ],
      },
    },
    "plugins": Array [
      "@babel/plugin-proposal-object-rest-spread",
    ],
    "presets": Array [
      Array [
        "@babel/env",
        Object {
          "targets": Object {
            "node": "12.19.0",
          },
        },
      ],
    ],
    "sourceMaps": true,
  },
  "devDependencies": Object {
    "@babel/cli": "^7.10.5",
    "@babel/core": "^7.11.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.11.0",
    "@babel/preset-env": "^7.11.0",
    "@babel/register": "^7.10.5",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "@babel/cli",
      "@babel/core",
      "@babel/register",
      "@babel/preset-env",
      "@babel/plugin-proposal-object-rest-spread",
    ],
  },
  "main": "index",
  "metapak": Object {
    "configs": Array [
      "babel",
      "mocha",
    ],
    "data": Object {
      "bundleFiles": Array [
        "dist",
        "src",
      ],
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "module": "index.mjs",
  "scripts": Object {
    "compile": "rimraf -f 'dist' && npm run compile:cjs && npm run compile:mjs",
    "compile:cjs": "babel --env-name=cjs --out-dir=dist --source-maps=true src",
    "compile:mjs": "babel --env-name=mjs --out-file-extension=.mjs --out-dir=dist --source-maps=true src",
    "precz": "npm run compile",
    "preversion": "npm run compile",
    "test": "",
  },
}
`;

exports[`Babel Package transformer should work with some files 1`] = `
Object {
  "babel": Object {
    "env": Object {
      "cjs": Object {
        "comments": true,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": "commonjs",
              "targets": Object {
                "node": "10",
              },
            },
          ],
        ],
      },
      "mjs": Object {
        "comments": false,
        "presets": Array [
          Array [
            "@babel/env",
            Object {
              "modules": false,
              "targets": Object {
                "node": "12",
              },
            },
          ],
        ],
      },
    },
    "plugins": Array [
      "@babel/plugin-proposal-object-rest-spread",
    ],
    "presets": Array [
      Array [
        "@babel/env",
        Object {
          "targets": Object {
            "node": "12.19.0",
          },
        },
      ],
    ],
    "sourceMaps": true,
  },
  "devDependencies": Object {
    "@babel/cli": "^7.10.5",
    "@babel/core": "^7.11.1",
    "@babel/plugin-proposal-object-rest-spread": "^7.11.0",
    "@babel/preset-env": "^7.11.0",
    "@babel/register": "^7.10.5",
  },
  "greenkeeper": Object {
    "ignore": Array [
      "@babel/cli",
      "@babel/core",
      "@babel/register",
      "@babel/preset-env",
      "@babel/plugin-proposal-object-rest-spread",
    ],
  },
  "main": "index",
  "metapak": Object {
    "configs": Array [
      "babel",
    ],
    "data": Object {
      "bundleFiles": Array [
        "src/**/*.js",
        "LICENSE",
        "README.md",
        "CHANGELOG.md",
        "dist/**/*.js",
        "dist",
        "src",
      ],
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "module": "index.mjs",
  "scripts": Object {
    "compile": "rimraf -f 'dist' && npm run compile:cjs && npm run compile:mjs",
    "compile:cjs": "babel --env-name=cjs --out-dir=dist --source-maps=true src",
    "compile:mjs": "babel --env-name=mjs --out-file-extension=.mjs --out-dir=dist --source-maps=true src",
    "precz": "npm run compile",
    "preversion": "npm run compile",
    "test": "",
  },
}
`;
